### DISCUSSION TITLE
Processor-level vfile message reporting

### DISCUSSION THREAD

**Discussion by dmca-glasgow** [unifiedjs/unified#243] [Ideas] [answered]
Hello,

This idea sort of evolved overnight after asking for advice about implementing some kind of [progress bar](https://github.com/orgs/unifiedjs/discussions/242).

I see that `unifiedjs/unified` uses the `vfile` dependency to [wrap a string input](https://github.com/unifiedjs/unified/blob/main/lib/index.js#L1284), but it could be useful if it added some processor-related messages about the order things happen, for example:

```
<timestamp> parse phase: remark-parse started
<timestamp> parse phase: remark-directive started
<timestamp> parse phase: remark-directive complete
<timestamp> parse phase: remark-parse complete
<timestamp> AST phase: my-custom-plugin started
<timestamp> AST phase: my-custom-plugin complete
<timestamp> AST phase: remark-rehype started
<timestamp> AST phase: remark-rehype complete
<timestamp> compile phase: rehype-stringify started
<timestamp> compile phase: rehype-stringify complete
```

This is something that would have helped me a lot when I started using unified, to understand that some plugins run in order while some don't, depending on their configuration.  

Together with a timestamp, this could also be used for performance analysis and optimisation, like generating flame charts.  

And it could also be the mechanism I could use to show useful information to the user on what's happening with their long running process.

**Comment by wooorm** [comment, marked as answer]
> [wrap a string input](https://github.com/unifiedjs/unified/blob/main/lib/index.js#L1284)

It’s better to think about it the other way:
unified accepts vfiles and gives back.
passing a string is a shortcut for passing a vfile.

---

I don’t think we should add such debugging messages: `unified` is used by *billions*. This would make it slower and bigger for *everyone*.

This seems quite similar to your Q in 242? This is something that people can do themselves if they want to by injecting a plugin between each plugin.

**Reply by dmca-glasgow** [reply]
> This would make it slower and bigger for everyone.

Fair enough.

> This is something that people can do themselves if they want to by injecting a plugin between each plugin.

Ahh ok, it's good to know that this approach is a valid one, I will adventure down this route :)
